core/src/main/java/com/linecorp/armeria/server/RoutingPredicate.java

```diff
@@ -108,37 +114,124 @@
+                                                                            BiFunction<U, String, Predicate<T>>
+                                                                                    equalsPredicate) {
+        final Matcher orMatcher = OR_PATTERN.matcher(predicateExpr);
+        if (orMatcher.find()) {
+            checkArgument(!predicateExpr.endsWith("|"), "Invalid predicate: %s (expected: '%s')",
+                          predicateExpr, TRAILING_PIPE_PATTERN.matcher(predicateExpr).replaceAll(""));
+            return Optional
+                    .of(Arrays
+                                .stream(OR_PATTERN.split(predicateExpr))
+                                .map(String::trim)
+                                .map(expression -> {
+                                    final Optional<NamedPredicate<T>> namedPredicate =
+                                            buildSingleExprNamedPredicate(expression,
+                                                                          nameConverter,
+                                                                          containsPredicate,
+                                                                          equalsPredicate);
+                                    checkArgument(namedPredicate.isPresent() && !expression.isEmpty(),
+                                                  "Invalid predicate: %s. Expression: %s (expected: '%s' or " +
+                                                          "'%s')",
+                                                  predicateExpr,
+                                                  expression,
+                                                  CONTAIN_PATTERN.pattern(),
+                                                  COMPARE_PATTERN.pattern());
+                                    return namedPredicate.get();
+                                })
+                                .reduce(NamedPredicate.empty(), NamedPredicate::or)
+                    );
+        }
+        return Optional.empty();
+    }
+
+    private static <T, U> Optional<NamedPredicate<T>> buildSingleExprNamedPredicate(String predicateExpr,
+                                                                                    Function<String, U>
+                                                                                            nameConverter,
+                                                                                    Function<U, Predicate<T>>
+                                                                                            containsPredicate,
+                                                                                    BiFunction<U, String,
+                                                                                            Predicate<T>>
+                                                                                            equalsPredicate) {
```

Comment:
How about adding a new line after `...NamedPredicate(`?
```suggestion
    @Nullable
    private static <T, U> NamedPredicate<T> buildSingleExprNamedPredicate(
            String predicateExpr, Function<String, U> nameConverter,
            Function<U, Predicate<T>> containsPredicate,
            BiFunction<U, String, Predicate<T>> equalsPredicate) {
```